discovery-infra/test_infra/controllers/nat_controller.py

```diff
@@ -0,0 +1,44 @@
+import logging
+from test_infra.utils import run_command
+
+class NatController:
+    def _get_default_interfaces(self):
+        interfaces, _, _ = run_command(r"ip -4 route | egrep '^default ' | awk '{print $5}'", shell=True)
+        return set(interfaces.strip().split())
+
+    def _build_mark_string(self, input_interface):
+        rule_template = ["PREROUTING", "-i", input_interface, "-j", "MARK", "--set-mark", "555"]
+
+        return " ".join(rule_template)
+
+    def _build_nat_string(self, output_interface):
+        rule_template = ["POSTROUTING", "-m", "mark", "--mark", "555", "-o", output_interface, "-j", "MASQUERADE"]
+
+        return " ".join(rule_template)
+
+    def _build_rule_string(self, option, rule_suffix):
+        rule_template = ["iptables", "-t", "nat",  f"--{option}", rule_suffix]
+
+        return " ".join(rule_template)
+
+    def _does_rule_exist(self, rule_suffix):
+        check_rule = self._build_rule_string('check', rule_suffix)
+        _, _, exit_code = run_command(check_rule, shell=True, raise_errors=False)
+
+        return exit_code == 0
+
+    def _insert_rule(self, rule_suffix):
+        insert_rule = self._build_rule_string('insert', rule_suffix)
+        logging.info("Adding rule \"%s\"", insert_rule)
+        run_command(insert_rule, shell=True)
+
+    def _add_rule(self, rule_suffix):
+        if not self._does_rule_exist(rule_suffix):
+            self._insert_rule(rule_suffix)
+
+    def add_nat_rules(self, input_interfaces):
```

Comment:
will this work in e2e test in parallel mode?

Reply:
maybe we should delete created rules too?

Reply:
It will work in parallel mode

Reply:
@tsorya where can we delete them? Currently I added a target that removes them in Makefile destroy target

Reply:
i mean in e2e? we will need to delete them right? or all none platform tests shouldn't be run in parallel ?

Reply:
The nat has two parts: The first is handle the input interfaces which is per cluster.  The second is the output interface(s) which is on host level.  I can try to remove the handling on cluster level.  destroy will remove all.